### PR TITLE
Use git to test solutions

### PR DESCRIPTION
**Pull request by nikandfor**

### CODE REVIEW

**Comment by rohitpaulk** [comment]
@nikandfor could you change the test workflow so that tests pass here? I think you'll just need to install `git` and move it to the expected secret location.

File: internal/stage_write_tree.go [hunk 1]
```diff
@@ -81,37 +56,120 @@ func testWriteTree(stageHarness *tester_utils.StageHarness) error {
...
+		return err
 	}
 
-	expectedValues := []string{rootFile, rootDir1, rootDir2}
-	sort.Strings(expectedValues)
-	expected := strings.Join(
-		expectedValues,
-		"\n",
-	) + "\n"
+	// check file list first as it's more useful
```
Comment (rohitpaulk) [inline internal/stage_write_tree.go:107]: Great call! 👌

**Comment by nikandfor** [comment]
As tests are done on local machine too, I changed git finding strategy. It first tries secret location and then the usual.

It's not testers responsibility to hide the git. But it's fine if tester find it anywhere.

**Comment by nikandfor** [comment]
Am I right that now I need to add git to each of these docker files?
image

**Comment by rohitpaulk** [comment]
> Am I right that now I need to add git to each of these docker files?

@nikandfor yep! And please make sure it is only accessible in the secret location and isn't in the PATH.

**Review by rohitpaulk — APPROVED** [review]

**Comment by rohitpaulk** [comment]
Feel free to merge this - that way we can make a release (`make release`) and test it on this PR: https://github.com/codecrafters-io/build-your-own-git/pull/21.

**Pull request merged**